### DISCUSSION TITLE
Auth-helpers in Next - useUser() help

### DISCUSSION THREAD

**Discussion by strangecarr** [supabase/supabase#9028] [General]
Hi, I've been trying to use auth-helpers, but I can't get useUser() to return anything except null. I'm using the Auth widget (which sits inside my AuthComponent), and when authorising (with google) I can see the session in local storage but useUser() isn't returning anything...

Am using:
        "@supabase/auth-helpers-nextjs": "^0.2.8",
        "@supabase/auth-helpers-react": "^0.2.4",
        "@supabase/supabase-js": "^1.35.6",

In my main _app.tsx, I have wrapped with UserProvider:

```
export default function App({ Component, pageProps }: AppProps) {
  return (
    <UserProvider supabaseClient={supabaseClient}>
          <Head>
            <meta name="viewport" content="width=device-width, height=device-height, initial-scale=1" />
          </Head>
            <Component {...pageProps} />
        </UserProvider>
  )
}
```

And from my Auth page (the AuthComponent renders, even when the user is authorised):

```
import { useState, useEffect } from 'react'
import { supabase } from '../utils/supabaseClient'
import { useUser } from '@supabase/auth-helpers-react'

import AuthComponent from '../components/Auth'
import Layout from '../components/Layout'

import { Button } from '@mui/material'


export default function Home() {
  const { isLoading, user, error } = useUser()

  if (!user)
  return (
    <Layout>
    <div className="container" >

      {!user ? (
        
          <AuthComponent />
       
      ) : 
      <div className="p-5">
        <h2>Looks like you are already signed in</h2>
        <Button variant="contained" color="primary" size="large" onClick={() => supabase.auth.signOut()}>
      Sign out
    </Button>
    </div>
    }       
    </div>
    </Layout>
  )
}
```

**Comment by saltcod** [comment]
Hey @strangecarr 

Wondering from a quick look here if the logic is right. 

Here's an alternate example — if `!user` render the form, otherwise, render the page as you'd expect. 
https://github.com/supabase/auth-helpers/blob/main/examples/nextjs/pages/index.tsx#L6

While you're setting this up from scratch now, you might also look into using supabase-js v2, so you won't need to rewrite any of this flow in a few months.   https://supabase.com/blog/supabase-js-v2
